File: python/stfmm3d_example.py
#!/usr/bin/env python

import fmm3dpy as fmm
import numpy as np


#
#  This is a sample code to demonstrate how to use
#  the fmm libraries
#

# sample with one density, sources to sources,
# stokeslet interactions, and potential+pressure+gradient
#
n = 20000
nd = 1
sources = np.random.uniform(0,1,(3,n))
eps = 10**(-5)

stoklet = np.random.uniform(0,1,(3,n)) 
out = fmm.stfmm3d(eps=eps,sources=sources,stoklet=stoklet,ifppreg=3)


# sample with a vector of densities, sources to 
# sources and targets, stokeslet + stresslet interactions, 
# potential + pressure

nd = 2
nt = 1870
targ = np.random.uniform(0,1,(3,nt))
stoklet = np.random.uniform(0,1,(nd,3,n)) 
strsvec = np.random.uniform(0,1,(nd,3,n)) 
strslet = np.random.uniform(0,1,(nd,3,n)) 
out = fmm.stfmm3d(eps=eps,sources=sources,stoklet=stoklet,\
    strslet=strslet,strsvec=strsvec,\
    targets=targ,nd=nd,ifppreg=2,ifppregtarg=2)

File: matlab/stfmm3d_example.m
disp("This is an example matlab driver for the stokes fmm");
disp("On output the code prints the pot+pre+grad in example1 and, pot+pottarg+pre+pretarg in example2");
disp(" ");
disp(" ");
%%   source to source, stokeslet only, pot+pre+grad example
%

clear srcinfo ns U eps pg;
disp("Example 1: source to source, stokeslet, pot+pre+grad");
disp(" ");
disp(" ");

ns = 40000;
srcinfo.sources = rand(3,ns);
srcinfo.stoklet = rand(3,ns);

eps = 1e-5;
ifppreg = 3;
U = stfmm3d(eps,srcinfo,ifppreg);
disp("pot=");
disp(U.pot(:,1:9));
disp("pre=");
disp(U.pre(1:9));
disp("grad=");
disp(U.grad(:,:,1:3));

clear srcinfo ns U eps ifppreg;
%
%%    source to source+target, stokeslet+stresslet, pot + pressure example
%
disp(" ");
disp(" ");
disp("Example 2: source to source+target, stokeslet + stresslet, multiple densities, pot+pressure");
disp(" ");
disp(" ");

ns = 4000;
nt = 3999;

nd = 2;
srcinfo.nd = nd;

ifppreg = 2;
ifppregtarg = 2;

srcinfo.sources = rand(3,ns);
srcinfo.stoklet = rand(nd,3,ns);
srcinfo.strslet = rand(nd,3,ns);
srcinfo.strsvec = rand(nd,3,ns);

targ = rand(3,nt);
eps = 1e-5;

U = stfmm3d(eps,srcinfo,ifppreg,targ,ifppregtarg);
disp("pot=");
disp(U.pot(:,:,1:3));
disp("pre=");
disp(U.pre(:,1:3));
disp("pottarg=");
disp(U.pottarg(:,:,1:3));
disp("pretarg=");
disp(U.pretarg(:,1:3));

clear srcinfo ns U eps ifppreg ifppregtarg nt targ nd;
